.github/workflows/spell-checker.yml

```
@@ -1,25 +1,33 @@
-name: Fix typos nightly
+name: Spell Check
 
 on:
-  workflow_dispatch: {} # enable manual running of workflow.
+  pull_request:
   schedule:
-    - cron: "0 8 * * *"
+    - cron: "0 12 * * *"
+
+permissions:
+  contents: write
+  pull-requests: write
 
 jobs:
   build:
     runs-on: ubuntu-latest
 
     steps:
       - uses: actions/checkout@v4
-      - uses: sobolevn/misspell-fixer-action@26173e641ba08255d0c9826b8a873f8b96f76932
+      - name: Run codespell
+        continue-on-error: true
+        run: |
+          sudo apt-get install codespell -y
+          codespell -w --skip="*.pb.go,*.pb.gw.go,*.json,*.git,*.js,*.h,*.bin,go.sum,go.mod,.github"
       - uses: peter-evans/create-pull-request@v7
+        if: github.event_name != 'pull_request'
         with:
-          token: ${{ secrets.GITHUB_TOKEN }}
-          commit-message: "chore: fixes by misspell-fixer"
-          title: "chore: typos fix by misspell-fixer"
+          commit-message: "chore: spelling errors fixes"
+          title: "chore: fix spelling errors"
           branch: "bot/fix-typos"
           delete-branch: true
           body: |
             This PR fixes typos in the codebase.
-            Please, review it and merge if everything is fine.
-            If there are proto changes, please, run `make proto-all` and commit changes.
+            Please review it, and merge if everything is fine.
+            If there are proto changes, run `make proto-gen` and commit the changes.
```

Probably best to keep `make proto-all` as it lints the protos and such as well :) 

```suggestion
            If there are proto changes, run `make proto-all` and commit the changes.
```

---

Updated @damiannolan